include/osquery/query.h

```diff
@@ -300,6 +304,14 @@ struct ScheduledQuery : private only_movable {
   ScheduledQuery(ScheduledQuery&&) = default;
   ScheduledQuery& operator=(ScheduledQuery&&) = default;
 
+  inline bool isSnapshotQuery() const {
+    return options.count("snapshot") && options.at("snapshot");
```

Comment:
This is two look up in hash table for the same key. You could do it with only one.
Would you like to use something like this?
```c++
auto it = options.find("snapshot");
return it != options.end() && *it;
```

Reply:
Could you proof somehow it please?

Because as far as I know:
1. Iterator is very lightweight class.
2. In most implementations `count` implemented via `find` : [libcxx implementation](https://llvm.org/svn/llvm-project/libcxx/trunk/include/__hash_table)
3. In most implementations `at` implemented via `find` : [libcxx implementation](https://llvm.org/svn/llvm-project/libcxx/trunk/include/unordered_map)

Reply:
Oh right :man_facepalming: 
I just rechecked myself.

I copied the original code but we should use `find()`.